marginalia.el

```diff
@@ -624,17 +624,21 @@ using `minibuffer-force-complete' on the candidate CAND."
 
 (defun marginalia-annotate-file (cand)
   "Annotate file CAND with its size, modification time and other attributes."
-  (when-let ((attributes (file-attributes (marginalia--full-candidate cand) 'string)))
-    (marginalia--fields
-     ((file-attribute-modes attributes) :face 'marginalia-file-modes)
-     ((format "%s:%s"
-              (file-attribute-user-id attributes)
-              (file-attribute-group-id attributes))
-      :width 12 :face 'marginalia-file-owner)
-     ((file-size-human-readable (file-attribute-size attributes)) :width 7 :face 'marginalia-size)
-     ((format-time-string
-       "%b %d %H:%M"
-       (file-attribute-modification-time attributes)) :face 'marginalia-date))))
+  (let ((win (active-minibuffer-window)))
+    (if (and win (file-remote-p (with-current-buffer (window-buffer win)
```

Comment:
I think you also need to check for the hosts completions as I have done in the other PR otherwise you will trigger opening connections for all the completions, see the late comments of the Selectrum issue.

Reply:
I don't understand. Why does my approach not work?

Reply:
I haven't tested it, maybe it works as you use completion-boundaries? `(file-remote-p "/ssh:c")` returns `nil` that was the reason I added the string-match checks for the hosts completion case.

Reply:
Okay. Could be. Maybe you find time to work on an improved PR later on? I don't have time to work on this right now.

Reply:
Sure I can try, maybe reconsider merging the other PR for now as the problem is quite drastic in effect when using the heavy annotators.

Reply:
No, this is not an urgent matter which justifies a quick fix. The issue has been present for a longer time now. There is also the possibility to deactivate heavy annotators for tramp.

Reply:
It opens connections to all completions, maybe just no one has complaint yet but sure we can wait if you think that is better.

Reply:
Is there a possibility that tramp keeps connections alive somehow? Does it really open a connection for each function call?

Reply:
Sorry that was poorly phrased, I meant when you do `/ssh:` it will open connections to all the completions you get for this (which can be previously connected hosts you are not connected to).

Reply:
Okay, makes sense. Now I understand what you meant with this host filtering above. Would be good to have this filtered here.